main.js

```diff
@@ -45,13 +44,81 @@ function generateNumbers() {
             allNumbers.push(randNumber);
         }
     }
-    while ((arrayCheckEqualsAsc(allNumbers)) || (arrayCheckEqualsDesc(allNumbers))) {
-        allNumbers = shuffleArray(allNumbers);
+
+    return allNumbers;
+}
+
+function generateNegativeNumbers(numbersWanted) {
+    let allNumbers = [];
+
+    for (let i = 0; i < numbersWanted; i++) {
+        let randNumber = Math.floor(Math.random() * 201) - 100;
+
+        if (allNumbers.includes(randNumber)) {
+            i--;
+        } else {
+            allNumbers.push(randNumber);
+        }
     }
 
     return allNumbers;
 }
 
+function generateHardNumbers(numbersWanted) {
+    let randNumber = '';
+    let allNumbers = [];
+    let numberSelect = Math.floor(Math.random() * 10);
+
+    for (let i = 0; i < numbersWanted; i++) {
+        let rand1in11chance = Math.floor(Math.random() * 11);
+
+        if (rand1in11chance === 10) {
+            randNumber = numberSelect;
+        } else {
+            let randSecondNumber = Math.floor(Math.random() * 10);
+
+            randNumber = parseInt(`${numberSelect}${randSecondNumber}`);
+        }
+
+        if (Math.random() < 0.5) {
+           randNumber = (0 - randNumber);
+        }
+
+        if (allNumbers.includes(randNumber)) {
```

Comment:
the potential same issue is there with the previous generateNumbers function (0-100) but its worse here as the range of potential values is significantly lowered (around 22 nubmers usually)